### ISSUE TITLE
bug(android/engine): Downloading keyboard updates in the background don't get preferred language

### ISSUE THREAD

**Issue by darcywong00** [keymanapp/keyman#7801]
Reported during testing of #7740 for 16.0 beta

When installing a keyboard package update (uses the background download manager), the installation process doesn't set the preferred language. So in the sil_cameroon_qwerty example, it ends up creating a new entry for the first language "Afade" in addition to the existing "Akum" language.

We should propagate the preferred language when installing keyboard updates.

**Comment by darcywong00** [comment]
Jotting some dev notes.
The API query for package updates returns a blob
```json
{
    "keyboards": {
        "sil_cameroon_qwerty": {
            "version": "6.0.8",
            "kmp": "https://keyman.com/go/package/download/keyboard/sil_cameroon_qwerty?version=6.0.8&update=1"
        }
    }
}
```

so maybe see if we need to append `&bcp47=...`

**Comment by darcywong00** [comment]
Some gotchas with sil_cameroon_qwerty.
| keyboard ID | package version | Akum language ID |
|---------------|-------------------|----------------------|
| sil_cameroon_qwerty | 6.0.4     |  aku-Latn                 |
| sil_cameroon_qwerty | 6.0.8     |  aku                         |

I believe since the language ID's don't fully match, the update will install with the first listed language (Afade - aal)

**Comment by darcywong00** [comment]
Fixed in 16.0 beta by #7807
